### PR TITLE
hw-2_watcher&importer

### PR DESCRIPTION
**Pull request by quantum-noise**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,12 +19,14 @@
   "homepage": "https://github.com/quantum-noise/njm#readme",
   "dependencies": {
     "babel-cli": "^6.26.0",
+    "babel-plugin-transform-class-properties": "^6.24.1",
     "babel-preset-env": "^1.6.0"
   },
   "devDependencies": {
     "babel-eslint": "^7.2.3",
     "eslint": "^4.6.1",
```
Comment (boccob) [inline package.json]: You have a bunch of errors, do you know about it?

File: workers/dirwatcher.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  process = (err, files) => {
+    if (!files) return;
+    const oldFiles = this.files;
+    const newFiles = [];
+    files.forEach(file => oldFiles.indexOf(file) === -1 && newFiles.push(file));
+    const newLength = newFiles.length;
+    if (newLength !== 0) {
+      this.files = oldFiles.concat(newFiles);
+      console.log(`Found ${newLength} new file${newLength > 1 && 's'}.`);
```
Comment (boccob) [inline workers/dirwatcher.js]: `Found 1 new filefalse.`. Could be better :)

File: index.js [hunk 3]
```diff
@@ -1,7 +1,21 @@
...
+const importer = new Importer(csvToJson, ENCODING);
+
+watcher.on(EVENTS_TYPE.changed, (path) => {
+  importer.import(path)
+    .then((data) => {
+      console.log(data);
+    });
+});
+
+setTimeout(() => console.log('Programm finished its work.') && watcher.close(), PROGRAMM_TIMEOUT);
```
Comment (boccob) [inline index.js]: Interesting idea but even after closing the watcher, the process is still running.

File: workers/dirwatcher.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    this.interval = setInterval(this.lookUp, delay);
+  }
+
+  static watch(path, delay) { // eslint-disable-line
+    return new DirWatcher(path, delay);
+  }
+
+  lookUp = () => fs.readdir(this.path, this.process);
+
+  process = (err, files) => {
```
Comment (boccob) [inline workers/dirwatcher.js]: But what if you have an error..

File: .babelrc [hunk 5]
```diff
@@ -5,5 +5,8 @@
                 "node": "current"
             }
         }]
+    ],
+    "plugins": [
+        "transform-class-properties"
```
[the remark this answers is not in the document]
Comment (quantum-noise) [reply .babelrc]: I need it for syntax like:
class A {
   b = () => {};
}

File: index.js [hunk 6]
```diff
@@ -1,7 +1,21 @@
 import appConfig from './config/app.json';
-import { User, Product } from './models';
+import DirWatcher, { EVENTS_TYPE } from './workers/dirwatcher';
+import Importer from './workers/importer';
+import csvToJson from './utils/cvsToJson';
 
-console.log(appConfig.name);
+const PROGRAMM_TIMEOUT = 25000;
```
[the remark this answers is not in the document]
Comment (quantum-noise) [reply index.js]: No =)

File: index.js (hunk 3 above)
Comment (quantum-noise) [reply index.js]: Yep, nodemon is still runing.

File: utils/cvsToJson.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+const cvsToJson = (data) => {
+  const lines = data.split('\n');
+  const keys = lines[0].split(',');
+  const arr = [];
+  for (let i = 1; i < lines.length; i++) {
```
[the remark this answers is not in the document]
Comment (quantum-noise) [reply utils/cvsToJson.js]: Got it

File: package.json (hunk 1 above)
Comment (quantum-noise) [reply package.json]: Cheked

File: workers/dirwatcher.js (hunk 4 above)
Comment (quantum-noise) [reply workers/dirwatcher.js]: Program prints it in console!

File: workers/dirwatcher.js (hunk 2 above)
Comment (quantum-noise) [reply workers/dirwatcher.js]: Done

File: index.js [hunk 8]
```diff
@@ -1,7 +1,21 @@
...
 
-const user = new User(); // eslint-disable-line
-const product = new Product(); // eslint-disable-line
+console.log(`Programm «${appConfig.name}» started.`);
+
+const watcher = DirWatcher.watch(appConfig.watchDir, 1000);
+const importer = new Importer(csvToJson, ENCODING);
+
+watcher.on(EVENTS_TYPE.changed, (path) => {
+  importer.import(path)
```
[the remark this answers is not in the document]
Comment (quantum-noise) [reply index.js]: Because it is hard enough concept for me at now.

**Pull request merged**